test/gulpfile.js

```diff
@@ -29,25 +29,36 @@ const gulp = help(_gulp);
 const testDir = __dirname;
 const apiTestDir = path.resolve(testDir, 'api');
 
-gulp.task('internal.test.clean.links', 'Delete npm links', () => {
-  return del([
-    path.resolve(testDir, 'node_modules/@grpc/js'),
-    path.resolve(testDir, 'node_modules/@grpc/native')
-  ]);
-});
-
 gulp.task('internal.test.install', 'Install test dependencies', () => {
   return execa('npm', ['install'], {cwd: testDir, stdio: 'inherit'});
 });
 
-gulp.task('internal.test.clean.all', 'Delete all files created by tasks',
-	  ['internal.test.clean.links']);
-
-gulp.task('internal.test.link.add', 'Link local copies of dependencies', () => {
-  linkSync(testDir, './node_modules/@grpc/js', '../packages/grpc-js');
-  linkSync(testDir, './node_modules/grpc', '../packages/grpc-native-core');
-});
+gulp.task('internal.test.clean.all', 'Delete all files created by tasks', () => {});
 
 gulp.task('internal.test.test', 'Run API-level tests', () => {
-  return gulp.src(`${apiTestDir}/*.js`).pipe(mocha({reporter: 'mocha-jenkins-reporter'}));
+  // run mocha tests matching a glob with a pre-required fixture,
+  // returning the associated gulp stream
+  const runTestsWithFixture = (glob, fixture) => new Promise((resolve, reject) => {
+    const server = fixture.split('_')[0];
+    const client = fixture.split('_')[1];
+    console.log(`Running ${glob} with ${server} server + ${client} client`);
+    gulp.src(glob)
+      .pipe(mocha({
+        reporter: 'mocha-jenkins-reporter',
+        require: `${testDir}/fixtures/${fixture}.js`
+      }))
+      .resume() // put the stream in flowing mode
+      .on('end', resolve)
+      .on('error', reject);
+  });
+  const apiTestGlob = `${apiTestDir}/*.js`;
+  const runTestsArgPairs = [
+    [apiTestGlob, 'native_native'],
```

Comment:
This scheme for specifying fixtures seems a bit backwards to me. Instead of writing the pair as one string and splitting it up, why not just write it like `{client: 'native', server: 'native'}` and join them if you have to?